homeassistant/components/zha/device_entity.py

```diff
@@ -108,13 +111,19 @@ def device_state_attributes(self):
             difference = time.time() - self._zha_device.last_seen
             if difference > self._keepalive_interval:
                 self._zha_device.update_available(False)
-                self._state = None
             else:
                 self._zha_device.update_available(True)
-                self._state = 'online'
                 if self._battery_listener:
                     await self.async_get_latest_battery_reading()
 
+    def async_set_available(self, available):
```

Comment:
Perhaps decorate this with `@callback` since it's part of the entity, so possible will be used by home assistant core at some future point.

Reply:
Will do